panic: Entry 10.32.0.0-10.36.0.1 reporting too much free space: 1048572 > 262145 - due to incorrect init-peer-count

Logs:
```
panic: Entry 10.32.0.0-10.36.0.1 reporting too much free space: 1048572 > 262145

goroutine 17 [running]:
github.com/weaveworks/weave/ipam/ring.(*Ring).assertInvariants(0xc2090e8180)
	/home/vagrant/src/github.com/weaveworks/weave/ipam/ring/ring.go:30 +0x92
github.com/weaveworks/weave/ipam/ring.(*Ring).Merge(0xc2090e8180, 0xa3000000a200000, 0x4a1bf2d34ef4, 0xc2090fafc0, 0x1, 0x1, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/ipam/ring/ring.go:258 +0x446
github.com/weaveworks/weave/ipam.(*Allocator).update(0xc209108000, 0x0, 0xc2093a8500, 0x252, 0x252, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/ipam/allocator.go:612 +0x44d
github.com/weaveworks/weave/ipam.func·011()
	/home/vagrant/src/github.com/weaveworks/weave/ipam/allocator.go:429 +0x4b
github.com/weaveworks/weave/ipam.(*Allocator).actorLoop(0xc209108000, 0xc20910a000)
	/home/vagrant/src/github.com/weaveworks/weave/ipam/allocator.go:473 +0xcf
created by github.com/weaveworks/weave/ipam.(*Allocator).Start
	/home/vagrant/src/github.com/weaveworks/weave/ipam/allocator.go:78 +0x79

goroutine 1 [chan receive]:
github.com/weaveworks/weave/common.SignalHandlerLoop(0xc2090cde40, 0x1, 0x1)
	/home/vagrant/src/github.com/weaveworks/weave/common/signals.go:20 +0x318
main.main()
	/home/vagrant/src/github.com/weaveworks/weave/releases/v1.1.0/prog/weaver/main.go:226 +0x335c

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 12 [select]:
github.com/weaveworks/weave/router.(*LocalPeer).actorLoop(0xc2090c3bf0, 0xc208062540)
	/home/vagrant/src/github.com/weaveworks/weave/router/local_peer.go:166 +0x101
created by github.com/weaveworks/weave/router.NewLocalPeer
	/home/vagrant/src/github.com/weaveworks/weave/router/local_peer.go:26 +0x13b

goroutine 14 [select]:
github.com/weaveworks/weave/router.(*Routes).run(0xc208062660, 0xc2080625a0, 0xc208062600)
	/home/vagrant/src/github.com/weaveworks/weave/router/routes.go:132 +0x132
created by github.com/weaveworks/weave/router.NewRoutes
	/home/vagrant/src/github.com/weaveworks/weave/router/routes.go:38 +0x415

goroutine 15 [select]:
github.com/weaveworks/weave/router.(*ConnectionMaker).queryLoop(0xc2090c1180, 0xc2080626c0)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection_maker.go:110 +0x162
created by github.com/weaveworks/weave/router.NewConnectionMaker
	/home/vagrant/src/github.com/weaveworks/weave/router/connection_maker.go:47 +0x187

goroutine 18 [select]:
github.com/fsouza/go-dockerclient.(*eventMonitoringState).monitorEvents(0xc208011340, 0xc208064e00)
	/home/vagrant/src/github.com/fsouza/go-dockerclient/event.go:176 +0x419
created by github.com/fsouza/go-dockerclient.(*eventMonitoringState).enableEventMonitoring
	/home/vagrant/src/github.com/fsouza/go-dockerclient/event.go:143 +0x16f

goroutine 19 [chan receive]:
github.com/weaveworks/weave/common/docker.func·001()
	/home/vagrant/src/github.com/weaveworks/weave/common/docker/client.go:42 +0x5c
created by github.com/weaveworks/weave/common/docker.(*Client).AddObserver
	/home/vagrant/src/github.com/weaveworks/weave/common/docker/client.go:49 +0x251

goroutine 20 [chan receive]:
github.com/weaveworks/weave/common/docker.func·001()
	/home/vagrant/src/github.com/weaveworks/weave/common/docker/client.go:42 +0x5c
created by github.com/weaveworks/weave/common/docker.(*Client).AddObserver
	/home/vagrant/src/github.com/weaveworks/weave/common/docker/client.go:49 +0x251

goroutine 21 [select]:
github.com/weaveworks/weave/nameserver.func·011()
	/home/vagrant/src/github.com/weaveworks/weave/nameserver/nameserver.go:71 +0x106
created by github.com/weaveworks/weave/nameserver.(*Nameserver).Start
	/home/vagrant/src/github.com/weaveworks/weave/nameserver/nameserver.go:78 +0x8c

goroutine 22 [syscall]:
syscall.Syscall(0x2f, 0x3, 0xc209158ab0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:21 +0x5
syscall.recvmsg(0x3, 0xc209158ab0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:1933 +0x57
syscall.Recvmsg(0x3, 0xc209152000, 0x200, 0x200, 0xc2091460c0, 0x28, 0x28, 0x0, 0xc2090e2280, 0x0, ...)
	/usr/local/go/src/syscall/syscall_linux.go:519 +0x19b
net.(*netFD).readMsg(0xc2090f41c0, 0xc209152000, 0x200, 0x200, 0xc2091460c0, 0x28, 0x28, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/fd_unix.go:292 +0x42b
net.(*UDPConn).ReadMsgUDP(0xc2090fa030, 0xc209152000, 0x200, 0x200, 0xc2091460c0, 0x28, 0x28, 0x4b689d0, 0x7f701629c4b0, 0x4168d1, ...)
	/usr/local/go/src/net/udpsock_posix.go:96 +0x17f
github.com/miekg/dns.ReadFromSessionUDP(0xc2090fa030, 0xc209152000, 0x200, 0x200, 0x200, 0x200, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/udp.go:47 +0xca
github.com/miekg/dns.(*Server).readUDP(0xc209128000, 0xc2090fa030, 0x77359400, 0x0, 0x0, 0x0, 0xb56f80, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/server.go:615 +0x14e
github.com/miekg/dns.(*defaultReader).ReadUDP(0xc209144020, 0xc2090fa030, 0x77359400, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/server.go:230 +0x72
github.com/miekg/dns.(*Server).serveUDP(0xc209128000, 0xc2090fa030, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/server.go:485 +0x1ea
github.com/miekg/dns.(*Server).ActivateAndServe(0xc209128000, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/server.go:355 +0x2cc
github.com/weaveworks/weave/nameserver.func·002(0xc209128000)
	/home/vagrant/src/github.com/weaveworks/weave/nameserver/dns.go:102 +0x28
created by github.com/weaveworks/weave/nameserver.(*DNSServer).ActivateAndServe
	/home/vagrant/src/github.com/weaveworks/weave/nameserver/dns.go:103 +0x74

goroutine 23 [IO wait]:
net.(*pollDesc).Wait(0xc2090f4290, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2090f4290, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc2090f4230, 0x0, 0x7f70162ad888, 0xc209150000)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2090fa038, 0xac0500, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
github.com/miekg/dns.(*Server).serveTCP(0xc2091280f0, 0xc2090fa038, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/server.go:446 +0x1bb
github.com/miekg/dns.(*Server).ActivateAndServe(0xc2091280f0, 0x0, 0x0)
	/home/vagrant/src/github.com/miekg/dns/server.go:360 +0x33c
github.com/weaveworks/weave/nameserver.func·002(0xc2091280f0)
	/home/vagrant/src/github.com/weaveworks/weave/nameserver/dns.go:102 +0x28
created by github.com/weaveworks/weave/nameserver.(*DNSServer).ActivateAndServe
	/home/vagrant/src/github.com/weaveworks/weave/nameserver/dns.go:103 +0x74

goroutine 33 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 34 [IO wait]:
net.(*pollDesc).Wait(0xc2080115d0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080115d0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208011570, 0xc2090e1000, 0x1000, 0x1000, 0x0, 0x7f70162ad888, 0xc209150080)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208036298, 0xc2090e1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc208062840)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).ReadSlice(0xc208062840, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:295 +0x257
net/http/internal.readLine(0xc208062840, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/internal/chunked.go:110 +0x5a
net/http/internal.(*chunkedReader).beginChunk(0xc20915e060)
	/usr/local/go/src/net/http/internal/chunked.go:47 +0x46
net/http/internal.(*chunkedReader).Read(0xc20915e060, 0xc209168000, 0x200, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/internal/chunked.go:77 +0xbb
net/http.(*body).readLocked(0xc20914c080, 0xc209168000, 0x200, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:584 +0x7a
net/http.(*body).Read(0xc20914c080, 0xc209168000, 0x200, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:579 +0x115
encoding/json.(*Decoder).readValue(0xc209166000, 0xc207eeb3e3, 0x0, 0x0)
	/usr/local/go/src/encoding/json/stream.go:124 +0x5e1
encoding/json.(*Decoder).Decode(0xc209166000, 0x979380, 0xc20914c1c0, 0x0, 0x0)
	/usr/local/go/src/encoding/json/stream.go:44 +0x7b
github.com/fsouza/go-dockerclient.func·008(0xc208082870, 0xc2080b2580)
	/home/vagrant/src/github.com/fsouza/go-dockerclient/event.go:285 +0x1b7
created by github.com/fsouza/go-dockerclient.(*Client).eventHijack
	/home/vagrant/src/github.com/fsouza/go-dockerclient/event.go:303 +0x625

goroutine 24 [runnable]:
github.com/weaveworks/weave/router.(*LocalPeer).ConnectionTo(0xc2090c3bf0, 0x4a1bf2d34ef4, 0xea6f, 0xc209410000, 0xea6f)
	/home/vagrant/src/github.com/weaveworks/weave/router/local_peer.go:90
github.com/weaveworks/weave/router.(*Router).udpReader(0xc2090c81a0, 0xc2090fa050, 0x7f7014830350, 0xc2090fa048)
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:250 +0x588
created by github.com/weaveworks/weave/router.(*Router).listenUDP
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:228 +0x324

goroutine 25 [IO wait]:
net.(*pollDesc).Wait(0xc2090f4370, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2090f4370, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc2090f4310, 0x0, 0x7f70162ad888, 0xc20908dfc0)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2090fa070, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
github.com/weaveworks/weave/router.func·031()
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:195 +0x5e
created by github.com/weaveworks/weave/router.(*Router).listenTCP
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:203 +0x2e4

goroutine 26 [syscall, locked to thread]:
github.com/google/gopacket/pcap._Cfunc_pcap_next_ex(0x2a0d3b0, 0xc2090ea3a8, 0xc2090ea3b0, 0x0)
	/home/vagrant/src/github.com/google/gopacket/pcap/:172 +0x43
github.com/google/gopacket/pcap.(*Handle).getNextBufPtrLocked(0xc2090ea380, 0xc209170ef8, 0x0, 0x0)
	/home/vagrant/src/github.com/google/gopacket/pcap/pcap.go:306 +0x92
github.com/google/gopacket/pcap.(*Handle).ZeroCopyReadPacketData(0xc2090ea380, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/vagrant/src/github.com/google/gopacket/pcap/pcap.go:338 +0xb0
github.com/weaveworks/weave/router.(*PcapIO).ReadPacket(0xc2090fa040, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/pcap.go:76 +0x67
github.com/weaveworks/weave/router.func·030()
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:131 +0x5d
created by github.com/weaveworks/weave/router.(*Router).sniff
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:136 +0x427

goroutine 38 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc209163020)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 37 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc209162fe0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 68 [chan receive]:
github.com/weaveworks/weave/router.(*Forwarder).run(0xc20914e140, 0xc2090dc2a0, 0xc2090dc300)
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:234 +0x89
created by github.com/weaveworks/weave/router.NewForwarder
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:216 +0x1b0

goroutine 52 [select]:
github.com/weaveworks/weave/router.(*LocalConnection).actorLoop(0xc208068d00, 0xc2080628a0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:462 +0x2e6
github.com/weaveworks/weave/router.(*LocalConnection).run(0xc208068d00, 0xc2080628a0, 0xc208062900, 0x7f701482c101)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:324 +0x672
created by github.com/weaveworks/weave/router.StartLocalConnection
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:113 +0x2fd

goroutine 66 [select]:
github.com/weaveworks/weave/router.(*LocalConnection).actorLoop(0xc2091ce000, 0xc2090dc180, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:462 +0x2e6
github.com/weaveworks/weave/router.(*LocalConnection).run(0xc2091ce000, 0xc2090dc180, 0xc2090dc1e0, 0x7f701482b701)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:324 +0x672
created by github.com/weaveworks/weave/router.StartLocalConnection
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:113 +0x2fd

goroutine 27 [IO wait]:
net.(*pollDesc).Wait(0xc209206c30, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc209206c30, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc209206bd0, 0x0, 0x7f70162ad888, 0xc2091e9408)
	/usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc2090fa418, 0x11b5dc0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc2090fa418, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc20920c240, 0x7f7014830238, 0xc2090fa418, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f7014830238, 0xc2090fa418, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1606 +0xa1
main.listenAndServeHTTP(0xc20908d640, 0x5, 0xc2090f0280)
	/home/vagrant/src/github.com/weaveworks/weave/releases/v1.1.0/prog/weaver/main.go:329 +0x2d9
created by main.main
	/home/vagrant/src/github.com/weaveworks/weave/releases/v1.1.0/prog/weaver/main.go:223 +0x3236

goroutine 56 [chan receive]:
github.com/weaveworks/weave/router.(*Forwarder).run(0xc2090c6550, 0xc208062cc0, 0xc208062d20)
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:234 +0x89
created by github.com/weaveworks/weave/router.NewForwarder
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:216 +0x1b0

goroutine 69 [select]:
github.com/weaveworks/weave/router.(*ForwarderDF).run(0xc2091be280, 0xc2090dc360, 0xc2090dc3c0, 0xc20935c000)
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:348 +0x47e
created by github.com/weaveworks/weave/router.NewForwarderDF
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:334 +0x2af

goroutine 57 [select]:
github.com/weaveworks/weave/router.(*ForwarderDF).run(0xc208065500, 0xc208062d80, 0xc208062de0, 0xc208086700)
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:348 +0x47e
created by github.com/weaveworks/weave/router.NewForwarderDF
	/home/vagrant/src/github.com/weaveworks/weave/router/forwarder.go:334 +0x2af

goroutine 70 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc2090de3e0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 71 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc2090de460)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 72 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc2090de4a0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 73 [semacquire]:
sync.(*RWMutex).RLock(0xc2090c1140)
	/usr/local/go/src/sync/rwmutex.go:36 +0x5f
github.com/weaveworks/weave/router.(*Peers).EncodePeers(0xc2090c1140, 0xc209147f20, 0x0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/peers.go:150 +0xba
github.com/weaveworks/weave/router.(*TopologyGossipData).Encode(0xc2090e53e0, 0x0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:373 +0x4f
github.com/weaveworks/weave/router.(*GossipChannel).sendBroadcast(0xc2090c6320, 0xc2a64fa2c7ea, 0x7f7014830e18, 0xc2090e53e0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip_channel.go:223 +0x16b
github.com/weaveworks/weave/router.func·016(0x7f7014830e18, 0xc2090e53e0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip_channel.go:209 +0x4a
github.com/weaveworks/weave/router.(*GossipSender).run(0xc2090de4e0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:61 +0xf6
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 74 [runnable]:
encoding/gob.(*decoderState).decodeUint(0xc209361180, 0xc209361180)
	/usr/local/go/src/encoding/gob/decode.go:151
encoding/gob.(*Decoder).decodeStruct(0xc208065a80, 0xc2093a49e0, 0xc2090e2dc0, 0xb04020, 0xc2093610c0, 0xd9)
	/usr/local/go/src/encoding/gob/decode.go:446 +0xaa
encoding/gob.(*Decoder).decodeValue(0xc208065a80, 0xc200000045, 0xb04020, 0xc2093610c0, 0xd9)
	/usr/local/go/src/encoding/gob/decode.go:1163 +0x477
encoding/gob.(*Decoder).DecodeValue(0xc208065a80, 0x97f380, 0xc2093610c0, 0x16, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decoder.go:210 +0x1d9
encoding/gob.(*Decoder).Decode(0xc208065a80, 0x97f380, 0xc2093610c0, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decoder.go:185 +0x297
github.com/weaveworks/weave/router.decodePeer(0xc208065a80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/vagrant/src/github.com/weaveworks/weave/router/peers.go:298 +0xc4
github.com/weaveworks/weave/router.(*Peers).decodeUpdate(0xc2090c1140, 0xc208039d40, 0x184, 0x184, 0xc20939d4d0, 0x16f0348, 0x0, 0x0, 0x16f0348, 0x0, ...)
	/home/vagrant/src/github.com/weaveworks/weave/router/peers.go:202 +0x21f
github.com/weaveworks/weave/router.(*Peers).ApplyUpdate(0xc2090c1140, 0xc208039d40, 0x184, 0x184, 0x41c7c9, 0x9751e0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/peers.go:110 +0x91
github.com/weaveworks/weave/router.(*Router).applyTopologyUpdate(0xc2090c81a0, 0xc208039d40, 0x184, 0x184, 0x16, 0xc209185518, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:401 +0x76
github.com/weaveworks/weave/router.(*Router).OnGossipBroadcast(0xc2090c81a0, 0x3212a25bca09, 0xc208039d40, 0x184, 0x184, 0x0, 0x0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/router.go:381 +0x7c
github.com/weaveworks/weave/router.(*GossipChannel).deliverBroadcast(0xc2090c6320, 0x3212a25bca09, 0xc20939a381, 0x1a2, 0x1a2, 0xc208065a00, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip_channel.go:80 +0x10c
github.com/weaveworks/weave/router.(*Router).handleGossip(0xc2090c81a0, 0xc208011906, 0xc20939a381, 0x1a2, 0x1a2, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip_channel.go:48 +0x32e
github.com/weaveworks/weave/router.(*LocalConnection).handleProtocolMsg(0xc208068d00, 0xc20939a306, 0xc20939a381, 0x1a2, 0x1a2, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:553 +0x28b
github.com/weaveworks/weave/router.(*LocalConnection).receiveTCP(0xc208068d00, 0x7f7014830960, 0xc2091842f0)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:532 +0x259
created by github.com/weaveworks/weave/router.(*LocalConnection).run
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:323 +0x652

goroutine 39 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc2091630a0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111

goroutine 40 [chan receive]:
github.com/weaveworks/weave/ipam.(*Allocator).OnGossip(0xc209108000, 0xc2093a8500, 0x252, 0x252, 0x0, 0x0, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/ipam/allocator.go:431 +0x398
github.com/weaveworks/weave/router.(*GossipChannel).deliver(0xc2090f0050, 0x4a1bf2d34ef4, 0xc2093a8001, 0x274, 0x274, 0xc2091d6700, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip_channel.go:92 +0xfc
github.com/weaveworks/weave/router.(*Router).handleGossip(0xc2090c81a0, 0xc2091bc004, 0xc2093a8001, 0x274, 0x274, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip_channel.go:50 +0x296
github.com/weaveworks/weave/router.(*LocalConnection).handleProtocolMsg(0xc2091ce000, 0xc2093a8004, 0xc2093a8001, 0x274, 0x274, 0x0, 0x0)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:553 +0x28b
github.com/weaveworks/weave/router.(*LocalConnection).receiveTCP(0xc2091ce000, 0x7f7014830960, 0xc209150240)
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:532 +0x259
created by github.com/weaveworks/weave/router.(*LocalConnection).run
	/home/vagrant/src/github.com/weaveworks/weave/router/connection.go:323 +0x652

goroutine 41 [select]:
github.com/weaveworks/weave/router.(*GossipSender).run(0xc2093acb20)
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:56 +0x22c
created by github.com/weaveworks/weave/router.NewGossipSender
	/home/vagrant/src/github.com/weaveworks/weave/router/gossip.go:50 +0x111
```

This code recreates the failure (when run within our weaveworks/weave/test environment)

````
#! /bin/bash

. ./config.sh

start_suite "Create inconsistency of IPAM rings"

weave_on $HOST1 launch-router --no-discovery
weave_on $HOST2 launch-router --no-discovery
weave_on $HOST3 launch-router --no-discovery

# Do some allocations to get the ring initialized on two hosts
weave_on $HOST1 expose
weave_on $HOST2 expose

# Connect 2 to 3 and allocate so they wil each have half the ring
weave_on $HOST2 connect $HOST3
weave_on $HOST3 expose

# More allocations on 1 to increase the version number
start_container $HOST1 --name=c1
start_container $HOST1 --name=c2

weave_on $HOST1 connect $HOST3

# Do something to see if HOST3 is still running
assert_raises "start_container $HOST3 --name=c3"

end_suite
````

Result (including some extra logging):
````
INFO: 2015/09/23 17:21:24.381370 ->[192.168.48.11:43888|f6:19:d2:5f:4a:9c(host1)]: connection added (new peer)
INFO: 2015/09/23 17:21:24.383715 ->[192.168.48.11:43888|f6:19:d2:5f:4a:9c(host1)]: Effective PMTU set to 65473
ERRO: 2015/09/23 17:21:24.386042 Error when merging entries: Entry 10.32.0.0-10.40.0.0 reporting too much free space: 1048573 > 524288
ERRO: 2015/09/23 17:21:24.386106 Previous entries: [0xc209369160 0xc209369180 0xc2093691a0]
ERRO: 2015/09/23 17:21:24.386208 Merge: {10.32.0.0 10.48.0.0 f6:19:d2:5f:4a:9c [0xc2093c7120]}
ERRO: 2015/09/23 17:21:24.386252 New state: Ring [10.32.0.0, 10.48.0.0)
  10.32.0.0 -> f6:19:d2:5f:4a:9c (v3)
  10.40.0.0 -> ce:3d:c8:69:c8:47 (v1)
  10.47.255.255 -> ca:63:f1:78:23:16 (v0)
panic: Entry 10.32.0.0-10.40.0.0 reporting too much free space: 1048573 > 524288

goroutine 15 [running]:
github.com/weaveworks/weave/ipam/ring.(*Ring).Merge(0xc2091020f0, 0xa3000000a200000, 0xf619d25f4a9c, 0xc2093ce290, 0x1, 0x1, 0x0, 0x0)
	/home/bryan/golang/src/github.com/weaveworks/weave/ipam/ring/ring.go:264 +0xa42
github.com/weaveworks/weave/ipam.(*Allocator).update(0xc2090d4240, 0x0, 0xc209388480, 0x22c, 0x22c, 0x0, 0x0)
	/home/bryan/golang/src/github.com/weaveworks/weave/ipam/allocator.go:623 +0x44d
github.com/weaveworks/weave/ipam.func·011()
	/home/bryan/golang/src/github.com/weaveworks/weave/ipam/allocator.go:440 +0x4b
github.com/weaveworks/weave/ipam.(*Allocator).actorLoop(0xc2090d4240, 0xc20805a720)
	/home/bryan/golang/src/github.com/weaveworks/weave/ipam/allocator.go:484 +0xcf
created by github.com/weaveworks/weave/ipam.(*Allocator).Start
	/home/bryan/golang/src/github.com/weaveworks/weave/ipam/allocator.go:78 +0x79
````

--- a/ipam/allocator.go
+++ b/ipam/allocator.go
@@ -618,16 +618,20 @@ func (alloc *Allocator) update(sender router.PeerName, msg []byte) error {
 	// shouldn't get updates for a empty Ring. But tolerate
 	// them just in case.
 	if data.Ring != nil {
-		if data.Ring.Range() != alloc.universe {
-			return fmt.Errorf("Incompatible IP allocation range %s; ours is %s",
-				data.Ring.Range().AsCIDRString(), alloc.universe.AsCIDRString())
+		switch err = alloc.ring.Merge(*data.Ring); err {
+		case ring.ErrDifferentSeeds:
+			return fmt.Errorf("IP allocation was seeded by different peers (received: %v, ours: %v)",
+				data.Ring.Seeds, alloc.ring.Seeds)
+		case ring.ErrDifferentRange:
+			return fmt.Errorf("Incompatible IP allocation ranges (received: %s, ours: %s)",
+				data.Ring.Range().AsCIDRString(), alloc.ring.Range().AsCIDRString())
+		default:
+			if err == nil && !alloc.ring.Empty() {
+				alloc.pruneNicknames()
+				alloc.ringUpdated()
+			}
+			return err
 		}
-		err = alloc.ring.Merge(*data.Ring)
-		if !alloc.ring.Empty() {
-			alloc.pruneNicknames()
-			alloc.ringUpdated()
-		}
-		return err
 	}
 
 	if data.Paxos != nil {

--- a/ipam/allocator.go
+++ b/ipam/allocator.go
@@ -354,6 +354,18 @@ func (alloc *Allocator) pruneNicknames() {
 	}
 }
 
+func (alloc *Allocator) annotatePeernames(names []router.PeerName) []string {
+	var res []string
+	for _, name := range names {
+		if nickname, found := alloc.nicknames[name]; found {
+			res = append(res, fmt.Sprint(name, "(", nickname, ")"))
+		} else {
+			res = append(res, name.String())
+		}
+	}
+	return res
+}
+
 func decodeRange(msg []byte) (r address.Range, err error) {
 	decoder := gob.NewDecoder(bytes.NewReader(msg))
 	return r, decoder.Decode(&r)
@@ -621,7 +633,7 @@ func (alloc *Allocator) update(sender router.PeerName, msg []byte) error {
 		switch err = alloc.ring.Merge(*data.Ring); err {
 		case ring.ErrDifferentSeeds:
 			return fmt.Errorf("IP allocation was seeded by different peers (received: %v, ours: %v)",
-				data.Ring.Seeds, alloc.ring.Seeds)
+				alloc.annotatePeernames(data.Ring.Seeds), alloc.annotatePeernames(alloc.ring.Seeds))
 		case ring.ErrDifferentRange:
 			return fmt.Errorf("Incompatible IP allocation ranges (received: %s, ours: %s)",
 				data.Ring.Range().AsCIDRString(), alloc.ring.Range().AsCIDRString())

--- a/ipam/ring/ring.go
+++ b/ipam/ring/ring.go
@@ -19,9 +19,10 @@ import (
 
 // Ring represents the ring itself
 type Ring struct {
-	Start, End address.Address // [min, max) tokens in this ring.  Due to wrapping, min == max (effectively)
-	Peer       router.PeerName // name of peer owning this ring instance
-	Entries    entries         // list of entries sorted by token
+	Start, End address.Address   // [min, max) tokens in this ring.  Due to wrapping, min == max (effectively)
+	Peer       router.PeerName   // name of peer owning this ring instance
+	Entries    entries           // list of entries sorted by token
+	Seeds      []router.PeerName // peers with which the ring was seeded
 }
 
 func (r *Ring) assertInvariants() {
@@ -36,6 +37,7 @@ var (
 	ErrNotSorted       = errors.New("Ring not sorted")
 	ErrTokenRepeated   = errors.New("Token appears twice in ring")
 	ErrTokenOutOfRange = errors.New("Token is out of range")
+	ErrDifferentSeeds  = errors.New("Received ring was seeded differently from ours!")
 	ErrDifferentRange  = errors.New("Received range differs from ours!")
 	ErrNewerVersion    = errors.New("Received new version for entry I own!")
 	ErrInvalidEntry    = errors.New("Received invalid state update!")
@@ -188,6 +190,17 @@ func (r *Ring) Merge(gossip Ring) error {
 		return err
 	}
 
+	if len(gossip.Seeds) > 0 && len(r.Seeds) > 0 {
+		if len(gossip.Seeds) != len(r.Seeds) {
+			return ErrDifferentSeeds
+		}
+		for i, seed := range gossip.Seeds {
+			if seed != r.Seeds[i] {
+				return ErrDifferentSeeds
+			}
+		}
+	}
+
 	if r.Start != gossip.Start || r.End != gossip.End {
 		return ErrDifferentRange
 	}
@@ -254,6 +267,9 @@ func (r *Ring) Merge(gossip Ring) error {
 		previousOwner = nil
 	}
 
+	if len(r.Seeds) == 0 {
+		r.Seeds = gossip.Seeds
+	}
 	r.Entries = result
 	return nil
 }
@@ -330,6 +346,8 @@ func (r *Ring) ClaimForPeers(peers []router.PeerName) {
 	}
 
 	common.Assert(pos == r.End)
+
+	r.Seeds = peers
 }
 
 func (r *Ring) FprintWithNicknames(w io.Writer, m map[router.PeerName]string) {

--- a/test/180_ipalloc_seed_mismatch_3.sh
+++ b/test/180_ipalloc_seed_mismatch_3.sh
@@ -1,0 +1,28 @@
+#! /bin/bash
+
+. ./config.sh
+
+start_suite "IPAM ring seed mismatch"
+
+weave_on $HOST1 launch-router --no-discovery
+weave_on $HOST2 launch-router --no-discovery
+weave_on $HOST3 launch-router --no-discovery
+
+# Do some allocations to get the ring initialized on two hosts
+start_container $HOST1
+start_container $HOST2
+
+# Connect 2 to 3 and allocate so they will each have half the ring
+weave_on $HOST2 connect $HOST3
+start_container $HOST3
+
+# More allocations on 1 to increase the version number
+start_container $HOST1
+start_container $HOST1
+
+weave_on $HOST1 connect $HOST3
+
+# Check that weave on 3 is still running
+assert_raises "weave_on $HOST3 status"
+
+end_suite

